src/plugins/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -185,10 +185,11 @@ function DiscoverDocumentsComponent({
 
   const showTimeCol = useMemo(
     () =>
-      !isTextBasedQuery &&
+      // for ES|QL we want to show the time column only when is on Document view
+      (!isTextBasedQuery || (isTextBasedQuery && !columns?.length)) &&
```

Comment:
Isn't implicit the positive `isTextBasedQuery` on the right handside?
```suggestion
      (!isTextBasedQuery || !columns?.length) &&
```